sos/report/plugins/openstack_ironic.py

```diff
@@ -20,41 +20,89 @@ class OpenStackIronic(Plugin):
-        ])
-
-        if self.get_option("all_logs"):
+
+        in_container = self.running_in_container()
+
+        if in_container:
+            self.conf_list = [
+                self.var_puppet_gen + "/etc/ironic/*",
+                self.var_puppet_gen + "/etc/ironic-inspector/*",
+                self.var_puppet_gen + "_api/etc/ironic/*",
+                self.ins_puppet_gen + "/etc/ironic-inspector/*",
+                self.ins_puppet_gen + "/var/lib/httpboot/inspector.ipxe"
+            ]
             self.add_copy_spec([
-                "/var/log/ironic/",
+                "/var/lib/ironic-inspector/",
+                "/var/log/containers/ironic-inspector/ramdisk/",
+                self.var_puppet_gen + "/etc/xinetd.conf",
+                self.var_puppet_gen + "/etc/xinetd.d/",
+                self.var_puppet_gen + "/etc/ironic/",
+                self.var_puppet_gen + "/etc/ironic-inspector/",
+                self.var_puppet_gen + "/etc/httpd/conf/",
+                self.var_puppet_gen + "/etc/httpd/conf.d/",
+                self.var_puppet_gen + "/etc/httpd/conf.modules.d/*.conf",
+                self.var_puppet_gen + "/etc/my.cnf.d/tripleo.cnf",
+                self.var_puppet_gen + "_api/etc/ironic/",
+                self.var_puppet_gen + "_api/etc/httpd/conf/",
+                self.var_puppet_gen + "_api/etc/httpd/conf.d/",
+                self.var_puppet_gen + "_api/etc/httpd/conf.modules.d/*.conf",
+                self.var_puppet_gen + "_api/etc/my.cnf.d/tripleo.cnf",
+                self.ins_puppet_gen + "/etc/ironic-inspector/*",
+                self.ins_puppet_gen + "/var/lib/httpboot/inspector.ipxe"
             ])
+
+            if self.get_option("all_logs"):
+                self.add_copy_spec([
+                    "/var/log/containers/ironic/",
+                    "/var/log/containers/ironic-inspector/"
+                ])
+            else:
+                self.add_copy_spec([
+                    "/var/log/containers/ironic/*.log",
+                    "/var/log/containers/ironic-inspector/*.log",
+                ])
+
+            for path in ['/var/lib/ironic', '/httpboot', '/tftpboot',
+                         self.ins_puppet_gen + '/var/lib/httpboot/',
+                         self.ins_puppet_gen + '/var/lib/tftpboot/']:
+                self.add_cmd_output('ls -laRt %s' % path)
+                self.add_cmd_output('ls -laRt %s' %
+                                    (self.var_puppet_gen + path))
         else:
+            self.conf_list = [
+                "/etc/ironic/*",
+                "/etc/ironic-inspector/*",
+            ]
             self.add_copy_spec([
-                "/var/log/ironic/*.log",
+                "/etc/ironic/",
+                "/etc/ironic-inspector/",
+                "/var/lib/ironic-inspector/",
+                "/var/log/ironic-inspector/ramdisk/",
+                "/etc/xinetd.conf",
+                "/etc/xinetd.d/",
```

Comment:
Capturing these from the `/var/log` location if running containerized is fine, but these are already captured by the `xinetd` plugin and should not be duplicated here.